nixos/modules/system/activation/activation-script.nix

```diff
@@ -208,7 +208,15 @@ in
       # "; true" => make the `$out` argument from switch-to-configuration.pl
       #             go to `true` instead of `echo`, hiding the useless path
       #             from the log.
-      default = "echo 'Warning: do not know how to make this configuration bootable; please enable a boot loader.' 1>&2; true";
+      default =
+        pkgs.writeShellScript
+          "no-bootloader"
+          "echo 'Warning: do not know how to make this configuration bootable; please enable a boot loader.' 1>&2";
+      defaultText = ''
```

Comment:
You want `lib.literalExpression` here. Also, a multi‐line string for the script would make it format nicer.

Reply:
I was actually doing this indentation specifically to avoid having to escape 2x single quotes in the defaultText :P but sure

Reply:
Fair.